tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,197 @@
+        key=Certificates.ECDSA_256.key,
+        cert=Certificates.ECDSA_256.cert,
+        trust_store=Certificates.ECDSA_256.cert,
+        insecure=False,
+        extra_flags=psk_params,
+        protocol=Protocols.TLS13)
+
+    server_psk_params = S2N_SERVER_PSK_PARAMETERS[psk_label]
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = S2N.ServerMode
+    server_options.extra_flags = server_psk_params
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    for results in client.get_results():
+        if psk_label == "single_psk_no_match":
+            assert b"Chosen PSK type: S2N_PSK_TYPE_EXTERNAL" not in results.stdout
+            assert b"Chosen PSK identity" not in results.stdout
+            assert results.exit_code == 0
+        else:
+            assert b"Chosen PSK type: S2N_PSK_TYPE_EXTERNAL" in results.stdout
+            assert bytes("Chosen PSK identity: {}".format(shared_psk_identity).encode('utf-8')) in results.stdout
+            assert results.exit_code == 0
+
+    for results in server.get_results():
+        if psk_label == "single_psk_no_match":
+            assert b"Chosen PSK type: S2N_PSK_TYPE_EXTERNAL" not in results.stdout
+            assert b"Chosen PSK identity" not in results.stdout
+            assert results.exit_code == 0
+            assert random_bytes in results.stdout
+        else:
+            assert b"Chosen PSK type: S2N_PSK_TYPE_EXTERNAL" in results.stdout
+            assert bytes("Chosen PSK identity: {}".format(shared_psk_identity).encode('utf-8')) in results.stdout
+            assert results.exit_code == 0
+            assert random_bytes in results.stdout
+
+pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("client_psk_params", S2N_CLIENT_PSK_PARAMETERS.items(), ids=get_parameter_name)
+def test_external_psk_s2nc_with_openssl_server(managed_process, client_psk_params):
+    port = next(available_ports)
+    random_bytes = data_bytes(64)
+    psk_label,psk_params = client_psk_params
+
+    client_options = ProviderOptions(
+        mode=S2N.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=Ciphers.AES128_GCM_SHA256,
+        data_to_send=random_bytes,
+        insecure=False,
+        key=Certificates.ECDSA_256.key,
+        cert=Certificates.ECDSA_256.cert,
+        trust_store=Certificates.ECDSA_256.cert,
+        extra_flags=psk_params,
+        protocol=Protocols.TLS13)
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = OpenSSL.ServerMode
+    server_options.extra_flags = OPENSSL_PSK_PARAMETERS
+    server = managed_process(OpenSSL, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    for results in client.get_results():
+        if psk_label == "single_psk_no_match":
```

Comment:
Why define results by test case name? Why not just include whether the test case is expected to be success / failure in the definition of the test case? Your test cases don't just have to be lists of parameters.